src/node/http.ts

```diff
@@ -57,20 +61,27 @@ export const ensureAuthenticated = (req: express.Request, _?: express.Response,
 /**
  * Return true if authenticated via cookies.
  */
-export const authenticated = (req: express.Request): boolean => {
+export const authenticated = async (req: express.Request): Promise<boolean> => {
   switch (req.args.auth) {
-    case AuthType.None:
+    case AuthType.None: {
       return true
-    case AuthType.Password:
+    }
+    case AuthType.Password: {
       // The password is stored in the cookie after being hashed.
-      return !!(
-        req.cookies.key &&
-        (req.args["hashed-password"]
-          ? safeCompare(req.cookies.key, req.args["hashed-password"])
-          : req.args.password && safeCompare(req.cookies.key, hash(req.args.password)))
-      )
-    default:
+      const hashedPasswordFromArgs = req.args["hashed-password"]
+      const passwordMethod = getPasswordMethod(hashedPasswordFromArgs)
+      const isCookieValidArgs: IsCookieValidArgs = {
+        passwordMethod,
+        cookieKey: sanitizeString(req.cookies.key),
+        passwordFromArgs: req.args.password || "",
+        hashedPasswordFromArgs: req.args["hashed-password"],
+      }
+
+      return await isCookieValid(isCookieValidArgs)
```

Comment:
Is there a reason we're passing the cookie stuff into a separate function instead of handling it in `authenticated`? All paths seem to lead here, so I don't know why we folded that out into a separate function,

Reply:
There are definitely multiple approaches one could take for this.

I went with this approach because it's:
- testable (rather than having to mock requests and test `authenticated`, though we will get there at one point)
- easy to read -> `isCookieValid` makes it very clear what the functionality is and what we're looking for
- maintainable -> should we move to a token approach, we can make changes to the underlying function and test in isolation

I completely understand if your style is to not extract things into separate functions. 

I will take this comment as a `nit` since it doesn't seem like there is much we're discussing besides the style/approach I took to solving the problem (unless I'm missing something).

Reply:
By the way we have login tests that make real requests so testing that should be pretty easy!

Reply:
Oh fantastic! I didn't even realize. That will be easy then!